packages/kit/src/core/sync/write_server.js

```diff
@@ -39,10 +42,11 @@ export const options = {
 	service_worker: ${has_service_worker},
 	templates: {
 		app: ({ head, body, assets, nonce }) => ${s(template)
-			.replace('%sveltekit.head%', '" + head + "')
-			.replace('%sveltekit.body%', '" + body + "')
+			.replace(/%sveltekit\.env\.([^%]+)%/g, (_match, capture) => public_env[capture] || '')
 			.replace(/%sveltekit\.assets%/g, '" + assets + "')
-			.replace(/%sveltekit\.nonce%/g, '" + nonce + "')},
+			.replace(/%sveltekit\.nonce%/g, '" + nonce + "')
+			.replace('%sveltekit.head%', '" + head + "')
+			.replace('%sveltekit.body%', '" + body + "')},
```

Comment:
Moved these to then end in case it helps to prevent undesired string replacement, eg the string `%sveltekit.nonce%` being included within the `head` or `body` output.

Reply:
No need! No replacement is happening at runtime, so there's no possibility of `%sveltekit.body%` containing content that gets erroneously replaced. I undid this so that the diff is clearer about what changed

Reply:
Thanks :D